Dewiz x Sidestream x Arby Spell Improvements

The-Arbiter opened the pull request:
Hi,

Changes as required from my 2023-05-10 Goerli review.

More to come as necessary.

SidestreamColdMelon commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -40,29 +41,34 @@ Spell Actions:
     * [ ] Match [ChainLog](https://chainlog.makerdao.com/)
   * [ ] Wards ok (pause proxy relied, deployer denied)
-    * [ ] Check whether the contract requires to rely the ESM in the spell (in order to allow de-authing the pause proxy during Emergency Shutdown, via `denyProxy`).
+    * [ ] `MCD_ESM` is relied (as approved by GovAlpha) in order to allow de-authing the pause proxy during Emergency Shutdown, via `denyProxy`.
```
I would suggest that those 6 occurrences of "`MCD_ESM` is relied" have to be publicly approved by the govalpha in this repo, not just casually [on discord](https://discord.com/channels/893112320329396265/897483518316265553/1105146557453574286) (which is semi-private)

SidestreamColdMelon replied:
FYI: this was already approved by govalpha here https://github.com/makerdao/pe-checklists/pull/12#issuecomment-1540017564

SidestreamColdMelon commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -39,7 +42,7 @@
         * [ ] ensure license is specified
         * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
-      * [ ] constructor arg matches [chainlog](https://chainlog.makerdao.com/)
+      * [ ] constructor arg matches [changelog](https://chainlog.makerdao.com/)
```
Nit: [changelog.makerdao.com](https://changelog.makerdao.com/) is the previous version of the contracts archive, stored on a regular server and on ipfs that was deprecated a year ago. So it's confusing to mix the terms and imply that it has to match deprecated service linking `chainlog.makerdao.com`
```suggestion
      * [ ] constructor arg matches [chainlog](https://chainlog.makerdao.com/)
```

SidestreamColdMelon replied:
If "constructor arg matches chainlog" means "constructor arg matches official ChainLog address", then it's correct, since `ChainLog` is the [name of the deployed contract](https://goerli.etherscan.io/address/0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F#code). Whatever it matches `getAddress("CHANGELOG")` or not it doesn't matter, since if it's unofficially deployed chainlog, `getAddress("CHANGELOG")` can return misleading information

SidestreamColdMelon reviewed (approved):
Changes are looking good from our side and doesn't contain anything we haven't discussed

patrick-j-govalpha commented:
For complete clarity this is the GovAlpha perspective:

- ES is a core component of the game theory in the MCD design used by the Maker Protocol (see: whitepaper).
- Failure to appropriately authorise the ESM on new contracts can lead to a failure mode in the event ES is called, with potential significant downsides for DAI holders as a result.
- We have recently paid out big bug bounties as a result of the identification of ES issues.
- It is, therefore, in my opinion, fine from a governance perspective to assume that the ESM should be authed for new contracts unless explicitly stated otherwise.

That said, things may change if, at some point, ES is removed from being a core component of the Protocol, and this should be reviewed if and/or when that happens.

patrick-j-govalpha commented:
For complete clarity this is the GovAlpha perspective:

- ES is a core component of the game theory in the MCD design used by the Maker Protocol (see: whitepaper).
- Failure to appropriately authorise the ESM on new contracts can lead to a failure mode in the event ES is called, with potential significant downsides for DAI holders as a result.
- We have recently paid out big bug bounties as a result of the identification of ES issues.
- It is, therefore, in my opinion, fine from a governance perspective to assume that the ESM should be authed for new contracts unless explicitly stated otherwise.

That said, things may change if, at some point, ES is removed from being a core component of the Protocol, and this should be reviewed if and/or when that happens.

SidestreamColdMelon commented:
A suggestion from `LongForWisdom | GovAlpha` [from the #new-spells discord channel](https://discord.com/channels/893112320329396265/897483518316265553/1108489040694607872): move [`Fill Spell Crafter Related Boxes in the Main Exec Doc Sheet` point](https://github.com/makerdao/pe-checklists/blob/492326ab00b4c400173b7d7d43a79df90c0c6c1d/spell/spell-crafter-mainnet-workflow.md?plain=1#L149) to much earlier in the process.

LongForWisdom commented:
So, there are three sets of things to confirm on the sheet.

1. (Contents sheet) 'Spellcrafter Confirm Content is feasible' - Ideally, this should be filled in shortly after GovAlpha creates the content sheet for the spell. Likely on Wednesday. Confirmation here indicates that the crafter understands and accepts the desired contents for the spell.
2. (Contents sheet) 'Spellcrafter confirm content in spell' - Ideally, this should be filled in after the mainnet spell is deployed? I guess? Open to feedback on when this should happen.
3. (Main sheet) The rows in green. We should discuss what should be included in the main sheet related to technical spell work. I suspect it's largely useless to you guys. For us it could be a good way to see what stage the spell is at without bothering you. 
 
The spell sheet itself has not been brought up to date recently. There are a number of rows that are no longer relevant, and the 'When?' column was never updated for the new schedule. We should discuss, and then GovAlpha should do an update pass on it.

The-Arbiter commented:
I agree. Let's sync up on this relatively soon so that we can balance overhead / effort expended with transparency. IMO we used to be a little slack on this due to the nature of content coming in late-ish or perhaps just due to the cadence. 

Having one "official spell comms channel" might be useful when it comes to these kinds of things too (re: you guys knowing where spells are at). Spell crafters determining feasibility of the content is a default yes (IMO) unless the content comes in late, since practically all types of content *can* be done, it's simply a matter of what gets done when. Pushback on here would rarely happen via sheet, I would assume. So you know, typically (obviously this will likely change with the introduction of subDAOs) all spell content is completed concurrently, with reviews (ideally) covering all included content at once in order to minimize the risk of something sneaking in through a later commit.

Happy to explore all the options available here with you and other spell participants at some point in the future; if you have any feedback or recommendations they are always appreciated!

The-Arbiter commented:
Considering enforcing that 1.00 rate (for SF) is called `ONE_PT_ZERO_ZERO_PCT_RATE` (i.e. we explicitly name trailing zeroes) given the move away from X.X5 and X.X0 type rates (divisible by 5 when in basis points) such as 3.49% this week (in order to keep names consistent and a predictable length and also to allow for simpler automation of spell checking).

Open to feedback on this

The-Arbiter commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/conduits/RwaSwapInputConduit2.sol) and it's consistent with previous RWA onboarding
       * [ ] constructor args are correct and match [chainlog](https://chainlog.makerdao.com/)
```
Should we add `quitTo` addresses here? Or are they too context dependent?

amusingaxl replied:
`quitTo` is meant to be used as an escape by the operator in case the PSM is full. If that happens, they can withdraw the stablecoins to the `quitTo` address using `quit()`.
Usually this address is the same as the `to` address for the output conduit, but that might not hold true every single time.

Worst case scenario, governance can always `yank` any gem in the input conduit, so in theory it's fine to not have this set.

Also notice that `quitTo` is `file`able, but it's not a constructor argument.

SidestreamSweatyPumpkin commented on spell/spell-crafter-goerli-workflow.md:
I would like to suggest an improvement to this checklist.

The bulletpoints regarding casting the spell are implying that the environment is fresh.
I've inferred this from https://github.com/foundry-rs/foundry/issues/4078 

Suggestion is to extend the casting segment with:
  - clean up / start a fresh shell environement
  - go set up the variables (inferring this from my experience of casting the spell for the first time, some mb not needed):
    - ETH_PASSWORD
    - ETH_KEYSTORE (file , not dir)
    - ETH_FROM
    - ETH_RPC_URL
    - ETHERSCAN_API_KEY

as per the link above - if we set the gas-related variables - it will break the script :(

The-Arbiter replied:
Opened an issue on `spells-goerli` for this, thanks for raising it.

The-Arbiter commented:
Let's merge this after this spell ships so we can be done with this PR and the content is on master :)

SidestreamColdMelon commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -170,7 +190,31 @@ Spell Actions:
+      * [ ] Any upgradable contracts have the `PAUSE_PROXY` as their `admin` (i.e. the party that can upgrade)
+        * [ ] Any upgradable SubDAO contracts with an `admin` that is not `PAUSE_PROXY` are not authed on any core contracts (Blocking)
+    * [ ] All SubDAO content addresses (i.e. provided contract addresses or EOAs) present in the Maker Core spell are present in the exec doc and are correct. SubDAO addresses being authed or given any permissions MUST be in the exec doc. SubDAO addresses being called must be confirmed by the SubDAO spell team.
+      * [ ] IF addresses not PR'ed in by the SubDAO team (use git blame for example), SubDAO content addresses all have inline comment for provenance or source being OKed by SubDAO
```
I'm not sure where is this requirement coming from. If SubDAO content is in the doc, why do we need this extra confirmation? How would then this confirmation look like? 

Suggestion: either remove it (preferred) or clearly specify what `OKed` means in very specific terms (eg. a comment in the PR)

The-Arbiter replied:
Yes sorry, this is a bit of a grey area.

Provenance and stuff are still not formalized but in the interim.

In my opinion, the following are OK:

Confirmation in Forum
Confirmation on GitHub via comment
Confirmation in Exec Sheet via comment
Discord Message saying it's OK (screenshot it)

This means 'written approval' in spirit.

Really this is a Governance Facilitator thing to handle so that's why it's a bit iffy.

This will be refined in the future, this is just for now.

The-Arbiter replied:
I also like this idea. 

However, can we do this at a separate time when we can get stakeholder involvement (SubDAO representatives as well as the subdao spell reviewers, which is who I assume you are referring to when you say 'the reviewers').

On the "code matches deployed address" stuff, this is outside of core spell team scope for a variety of technical reasons, as there is no infra in place like the extcodehash (soul) [check](https://github.com/dapphub/ds-pause/blob/master/src/pause.sol#L97) present for SubDAOs at present.

IMO this is a good point but should be tackled separately to avoid inflammation of what is already a sensitive issue.

Does this work for you?

SidestreamColdMelon replied:
Agreed that this can be discussed and done in a separate PR

SidestreamColdMelon commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -170,7 +190,31 @@ Spell Actions:
+        * [ ] Any upgradable SubDAO contracts with an `admin` that is not `PAUSE_PROXY` are not authed on any core contracts (Blocking)
+    * [ ] All SubDAO content addresses (i.e. provided contract addresses or EOAs) present in the Maker Core spell are present in the exec doc and are correct. SubDAO addresses being authed or given any permissions MUST be in the exec doc. SubDAO addresses being called must be confirmed by the SubDAO spell team.
+      * [ ] IF addresses not PR'ed in by the SubDAO team (use git blame for example), SubDAO content addresses all have inline comment for provenance or source being OKed by SubDAO
+    * [ ] SubDAO actions match exec doc (only where inline with main spell code) and do not affect core contracts
```
Let's clearly define two different terms for code in the core spell related to SubDAO (eg `SubDAO-related actions`) and code inside SubDAO spell (eg `actions inside SubDAO spell`) and then use them exclusively in the doc. Otherwise it's kind of a mess. 

From the current wording I don't get what you mean by `SubDAO actions` – SubDAO-related actions in the core spell are already covered in the bullet-points above, actions inside SubDAO spell are reviewed separately

The-Arbiter replied:
I agree, I will think about this more.

 SubDAO actions --> Subdao spell actions 
 SubDAO core actions --> Subdao requested changes to Maker Protocol contracts

Any feedback here?

SidestreamColdMelon replied:
Also, do you want to clearly state where responsibility ends with the SubDAO-spell reviews? Eg: `SubDAO-spell content has at least two public approvals and is linked in the spell`

The-Arbiter replied:
> Also, do you want to clearly state where responsibility ends with the SubDAO-spell reviews? Eg: `SubDAO-spell content has at least two public approvals and is linked in the spell`

Not in the checklist yet since this is still up in the air and needs to be formalized - see my reply to the earlier point on this.

I agree that this definition should be in the checklist in principle but the checklist isn't a source of truth for things like this, so maybe let's regroup and settle it with stakeholder (subdao) input separately?

> The wording is not very usable cause it's quite long. Maybe introducing subspell/SubDAO-spell term would help? Ie `SubDAO-spell actions` for something that SubDAO writes and `SubDAO-related actions` for something that is contained within the core spell? If you agree, please edit the wording of the checklist

SubDAO actions == "SubDAO-spell actions"
SubDAO core / SubDAO core actions == "SubDAO-related actions"

Here, the wording needs to be considered carefully since again this isn't the source of truth on this stuff. Soemthing can be subdao related but also not maker core and also not in a spell (it may not even go through the pause proxy), so I limited it to 'core' which only the PP can execute and then 'actions' which is just "!core" i.e. subdao spell and other subdao related stuff.

Limiting spell team coverage to subdao core rather than "subdao related" limits the scope of stuff being covered by the core spell team in a better way IMO and is a more strict definition ('core' is clearer than 'related').

Open to feedback (or alternative names!) here :)

SidestreamColdMelon commented on spell/teleport-onboarding-checklist.md:
```diff
@@ -27,6 +27,7 @@
           * [ ] others (L2 dependant)
         * [ ] check `wards`
           * [ ] `L2GovernanceRelay` is relied
+          * [ ] `MCD_ESM` is already relied / being relied this spell (as approved by GovAlpha)
```
```suggestion
          * [ ] `MCD_ESM` is already relied / being relied in this spell (as approved by GovAlpha)
```
(here and in two other places)

The-Arbiter replied:
Fixed

The-Arbiter commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] ensure license is specified
-        * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/conduits/RwaSwapOutputConduit.sol) and it's consistent with previous RWA onboarding
```
I put the commit hash here since it's what we are using and makes it easier for reviewers and crafters to check the source.

Otherwise you'd have to find the correct version every spell; you don't need to use the commit hash, but it's there if you want it.

amusingaxl replied:
BTW new conduits are being audited, so this shouldn't be an issue.

The-Arbiter commented on spell/spell-crafter-goerli-workflow.md:
Good spot.

I am using

```
* [ ] Create a new branch on the `spells-goerli` repo named `YYYY-MM-DD` using the initial target date of the spell
  * [ ] Ensure the same target date is used as the corresponding `spells-mainnet` spell branch
```

Reasoning:
`YYYY-MM-DD` name is consistent between mainnet and goerli repos
`YYYY-MM-DD` for different dates will sort when done alphabetically
`YYYY-MM-DD` makes sense when you read it as human language
`PE-XXXX YYYY-MM-DD Spell` was the PECU format for spell PR names, where XXXX was the Kanban ticket number (which we no longer have), which this does not vary much from.

Noting here that maintenance and improvement PRs should use a suffix such as 'Maintenance' or 'Backport' and emergency / urgent "Emergency" / "Urgent"

SidestreamColdMelon replied:
`YYYY-MM-DD` as branch name is fine with me

But on line 38 there are still `PE-<ticket number>: YYYY-MM-DD Executive` requirement https://github.com/makerdao/pe-checklists/blob/b412dcabe2e85a4b380f0da227276b640e179bf9/spell/spell-crafter-goerli-workflow.md?plain=1#L38

I suggest to change it to `YYYY-MM-DD executive spell` as captured in some of our discussions

amusingaxl replied:
This is for the PR title, isn't it?
PRs have an ID, so we can use that for sorting instead of the date itself.

The-Arbiter commented on .wordlist.txt:
```diff
@@ -23,9 +23,11 @@ Etherscan
 diffcheck
 diffchecked
 ChainLog
+changelog
```
Done

The-Arbiter replied:
Confirmed unused, removed.
Done

The-Arbiter commented on spell/spell-crafter-goerli-workflow.md:
```diff
@@ -74,6 +75,9 @@ PR: https://github.com/makerdao/spells-goerli/pull/TODO
   * [ ] Consider `immutable` visibility when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress` and use `constant` for static addresses
+    * [ ] Fetch addresses as type `address` and wrap with `Like` suffix interfaces inline (when making calls) unless archive patterns permit otherwise (Such as `MKR`)
+    * [ ] Use the [DssExecLib Core Address Helpers](https://github.com/makerdao/dss-exec-lib/blob/master/src/DssExecLib.sol#L166) where possible (e.g. `DssExecLib.vat()`)
+    * [ ] Variable names of addresses fetched from the `ChainLog` match the corresponding ChainLog key (e.g. `MCD_VAT` rather than `vat`) except where archive pattern differs (e.g. MKR)
```
Existing:

```* [ ] Use variable names of addresses fetched from the `ChainLog` match the corresponding ChainLog key (e.g. `MCD_VAT` rather than `vat`) except where archive pattern differs (e.g. MKR)```

New:

```If addresses are fetched from the `ChainLog`, the variable name must match the value of the ChainLog key for that address (e.g. `MCD_VAT` rather than `vat`), except where the archive pattern differs from this pattern (e.g. MKR)```

This refers to an action for the crafter and a check for the reviewer, which raises a different question - do we want actions that refer to the same thing to have different wording on the crafter and reviewer checklists or do we want the wording to be the same (as it currently is) at the cost of checklist items sometimes not being as readable in terms of grammar or syntax.

Of course the benefit of the existing method if that maintenance of the checklist is easier and it's easy to spot corresponding checklist items.

I will update this entry but this stuff poses a good question

The-Arbiter commented on spell/spell-crafter-mainnet-workflow.md:
```diff
@@ -78,6 +79,9 @@ Repo: https://github.com/makerdao/spells-mainnet
   * [ ] Consider `immutable` visibility when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress` and use `constant` for static addresses
+    * [ ] Fetch addresses as type `address` and wrap with `Like` suffix interfaces inline (when making calls) unless archive patterns permit otherwise (Such as `MKR`)
+    * [ ] Use the [DssExecLib Core Address Helpers](https://github.com/makerdao/dss-exec-lib/blob/master/src/DssExecLib.sol#L166) where possible (e.g. `DssExecLib.vat()`)
+    * [ ] Variable names of addresses fetched from the `ChainLog` match the corresponding ChainLog key (e.g. `MCD_VAT` rather than `vat`) except where archive pattern differs (e.g. MKR)
```
Fixed all instances at once

The-Arbiter commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -135,37 +143,40 @@ Spell Actions:
+    * [ ] Execution is NOT delegate call
+    * [ ] Gas cost will not be too high (low level call gas estimation is not done by our scripts)
+  * [ ] Maker Core (main spell) SubDAO actions (i.e. operate in Pause Proxy `DelegateCall` context)
+    * [ ] No SubDAO contract being interacted with is authed on a core contract like vat, etc. (script this eventually)
```
Fixed and changed to 'Check comprehensively where the risk is high'

Originally I put a TODO here since I'm not sure where else it would go and it's better that these things are internally consistent as a concerned / cautious reviewer can deep dive into the issue if they know this was an item of priority but they won't know that this was an area of concern if there is no note here.

That being said I agree TODOs in here are annoying and they should be removed where possible.

The "Check comprehensively where the risk is high" comment fulfils the original purpose while removing the issue.

Noting here that there is an open TODO item for an `auth` check script which can use a generic address as input

The-Arbiter commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -135,37 +143,40 @@ Spell Actions:
+    * [ ] Gas cost will not be too high (low level call gas estimation is not done by our scripts)
+  * [ ] Maker Core (main spell) SubDAO actions (i.e. operate in Pause Proxy `DelegateCall` context)
+    * [ ] No SubDAO contract being interacted with is authed on a core contract like vat, etc. (script this eventually)
+    * [ ] SubDAO contract licensing and optimizations do not matter (not strictly)
```
SubDAOs are already excluded from the base case `External Contracts`

`* [ ] External Contracts Calls (Not SubDAOs, e.g. Starknet)`

You're correct that message `* [ ] SubDAO contract licensing and optimizations do not matter (not strictly)` isn't really a check but is a note for reviewers, otherwise the standard procedure for checking smart contracts is missing a step.

Agree with your concerns here and will find a better solution in the future, though for now this is the only area where licensing of subdao contracts is covered

The-Arbiter commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -135,37 +143,40 @@ Spell Actions:
+        * [ ] Accompanying comment above `bump` line in format `// XXM * 1.XX^X * X.XX as a WAD` corresponding to the `val` calculation formula (e.g. `// 15M * 1.03^2 * 1.00 as a WAD`) is present along with the calculation formula on the line above
+        * [ ] IF combining `val` of multiple RWA ilks being combined, `val` calculation is done once per ilk and added to make the total, with workings provided in code comments. The existing `val` value can be retrieved by calling `read()` on `PIP_RWAXX` and converting the result into decimal.
     * [ ] Poke `spotter` to pull in the new price
-* [ ] Payments
-  * [ ] Streams (`DssVest`)
-    * [ ] `DssVest` Interface is correct
-    * [ ] Ensure that `cap` > max new vest `tot`/`tau` otherwise file cap as well
```
What do you mean by this? Payments here refers to the category of content (MKR payments DAI payments, vests, streams, yanking of vests, etc.)

SidestreamColdMelon replied:
I meant that you completely removed payments from the goerli checklist as if there is no rules about it. My suggestion is to at least mention `Payments` in some way of the other, eg:

```
* [ ] Payments
   * [ ] No actual transfers or streams is done on Goerli (but they can optionally be present as comments)
```

The-Arbiter replied:
Ah yes this is true, yes let me do that.

The-Arbiter commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -174,11 +185,18 @@ Spell Actions:
 * [ ] Ensure `immutable` visibility is only used when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress` and `constant` is used instead for static addresses
+  * [ ] Fetch addresses as type `address` and wrap with `Like` suffix interfaces inline (when making calls) unless archive patterns permit otherwise (Such as `MKR`)
+  * [ ] Use the [DssExecLib Core Address Helpers](https://github.com/makerdao/dss-exec-lib/blob/master/src/DssExecLib.sol#L166) where possible (e.g. `DssExecLib.vat()`)
+  * [ ] Variable names of addresses fetched from the `ChainLog` match the corresponding ChainLog key (e.g. `MCD_VAT` rather than `vat`) except where archive pattern differs (e.g. MKR)
```
Fixed

The-Arbiter commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -170,7 +190,31 @@ Spell Actions:
+    * [ ] Execution is NOT delegate call
+    * [ ] Gas cost will not be too high (low level call gas estimation is not done by our scripts)
+  * [ ] Maker Core (main spell) SubDAO actions (i.e. operate in Pause Proxy `DelegateCall` context)
+    * [ ] No SubDAO contract being interacted with is authed on a core contract like vat, etc. (script this eventually)
```
Fixed

The-Arbiter commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -170,7 +190,31 @@ Spell Actions:
+    * [ ] Gas cost will not be too high (low level call gas estimation is not done by our scripts)
+  * [ ] Maker Core (main spell) SubDAO actions (i.e. operate in Pause Proxy `DelegateCall` context)
+    * [ ] No SubDAO contract being interacted with is authed on a core contract like vat, etc. (script this eventually)
+    * [ ] SubDAO contract licensing and optimizations do not matter (not strictly)
```
Reworded it but see above

SidestreamColdMelon commented:
Few comments are still pending (sadly, I can't collapse resolved comments, as I don't have access to this repo):
- Goerli payments https://github.com/makerdao/pe-checklists/pull/12#discussion_r1276295250
- SubDAO link to PR https://github.com/makerdao/pe-checklists/pull/12#discussion_r1276324433
- SubDAO wording https://github.com/makerdao/pe-checklists/pull/12#discussion_r1276316897
- SubDAO responsibility https://github.com/makerdao/pe-checklists/pull/12#discussion_r1276316967

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -135,37 +132,53 @@ Spell Actions:
+* [ ] SubDAO Content
+  * [ ] SubDAO SubProxy spell execution
+    * [ ] SubDAO spell address matches exec doc
+    * [ ] SubDAO spell deployer is in `addresses_deployers`
```
I don't think we should have this requirement.

Should we care about who deploys a subdao spell?

I'm currently deep diving into the spell test suite, and the only time the addresses in `addresses_deployers.sol` are used is in the [`checkAuth` test](https://github.com/makerdao/spells-mainnet/blob/c1f3caa1374e4f911db6a246852947c2b76692b3/src/DssSpell.t.base.sol#L1529-L1545).

This test basically fetches all contracts in the chainlog and checks if all deployers have been `deny`ed in those.

Spells don't make it into the chainlog, so having the deployer of a spell there does not do anything other than maybe keeping a list there. This might be important for MakerDAO spells, for the sake of the spell-crafting process, but I don't think we should have SubDAO spells deployers there.

amusingaxl replied:
cc @SidestreamColdMelon @The-Arbiter @SidestreamSweatyPumpkin @0xdecr1pto

The-Arbiter replied:
If you're OK with at least documenting it somewhere I am OK with that given what you've mentioned here (and our discussions last week on this item).

> Should we care about who deploys a subdao spell?

Only if they can affect the core system / contracts we care about - which happens to be what this test checks.

That being said, it _does_ (parabolically at best) increase test execution time, so I think it isn't the end of the world to not check that Auth for these EOAs.

_However:_
If the address is not an EOA (it is a multisig, a factory, etc.) then I think we should check this.

I also think we should have a log of the deployer addresses so that we don't get attacked by a similar looking deployer address doing something, so there should at the very least be a log of which deployer addresses were used in the past.

If you look at the addresses_deployers.sol on Goerli, it is much more labelled than the one on Mainnet because people in the past decided that it wasn't necessary to label addresses for the mainnet branch, so that's the impetus behind my thoughts on this (I don't want to make that 'mistake' again).

Either way, the SubDAO actions were put together quickly and so this does seem like an extraneous (or at best, minimally helpful) check, so I will modify it a bit:

```
* [ ] SubDAO spell deployer is a smart contract (e.g. multisig or factory)
      * [ ] Ensure the deployer address is in `addresses_deployers.sol` as an entry
    * [ ] SubDAO spell deployer is purely an EOA
      * [ ] Ensure the deployer address is in `addresses_deployers.sol` as a comment
```

This solves the problems:

- No additional test running time
- Deployer address is still _transparently_ available (and accessible by third parties who wish to audit spells for whatever reason)
- Information regarding contract deployers is in one place (the addresses_deployers.sol file)

I still think that auth checks here are necessary for the following reason:

When the MakerDAO-level spell is calling 'execute' on something (not a DelegateCall), this can only be done comfortably because it is known that the spell contract is not authed on any core contracts. 
The only way to auth something on a core contract is to already be `relied` on said core contract.
One of the only ways that a spell being 'executed' can be authed on a core contract is if some address (such as an EOA) auths the spell, allowing it to perform potentially harmful actions.

Of course, the EOA could do it themself, which is why EOAs are (to my knowledge) never authed on core contracts within MCD, _however_ it is foreseeable that Multisigs for SubDAOs may be treated as more 'trusted' actors (which are not strictly EOAs if they're smart contracts) and therefore might have auth on something core (even if it's peripheral rather than something vital like Vat), so these checks might prevent something in those cases (obviously they aren't currently happening though).

Either way, this is reworked for now so that it does not impact test execution time while preserving transparency and documenting (potential) hazards.

SidestreamSweatyPumpkin commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -195,16 +246,20 @@ Spell Actions:
     * [ ] optimizations are disabled
-    * [ ] dss-exec-lib library address matches hardcoded local `DssExecLib.address`
+    * [ ] dss-exec-lib library address used (under 'Libraries Used') matches the hardcoded local `DssExecLib.address` file
+      * [ ] Check again that the PR did not modify the `DssExecLib.sol` file (e.g. look under the 'Files Changed' PR tab, etc.)
```
DssExecLib.address or sol?

The-Arbiter commented:
I renamed `Risk Parameter Changes` to `Core System Parameter Changes` in order to better reflect the nature of the checks performed within this section:

- DSR changes are not explicitly 'risk' associated changes (though this could be argued) and are more 'reward' oriented
- Other auction parameters such as starting price factor (`buf`) are also only somewhat 'risk' in my opinion.

A better name was 'Risk and Reward Parameter Changes' but the name 'Core System Parameter Changes' better reflects these items in this section as there are risk parameter changes (RWA, etc.) which occur outside of this section.

An alternative name might be 'Core System Risk Parameter Changes' but this is too wordy for my liking and `System Parameter Changes` was too general, so I went with `Core System Parameter Changes`.

_Anyone is free to provide feedback here or veto this name change if they feel it is inappropriate._

Noting that TypoCheck is failing but I will fix it before this PR gets merged.

The-Arbiter commented:
I removed the following lines:

```
* [ ] CI tests are passing
* [ ] Local Tests PASS

_Insert your passing local tests here_
```

Under the **Cast Stage** section for the following reasons:

- This is never complied with because _local tests should almost always fail after a spell is cast._ As such, these items were unused in the first place.
- This check originated from repeatedly adding 'tests are passing' style checks after various stages in the checklist. Lack of sections / headings within the checklist probably caused confusion and led to this being here.

_Anyone is free to veto this change as it is a functionality related change and removes the need to run tests._

The-Arbiter commented:
Made more changes:

- Added 'Per Exec Doc / Sheet' for Mainnet/Goerli respectively as part of 'Spell Actions' for clarity
- Added instruction for spell actions section to ensure compliance
- Renamed coding stage to development stage to accurately reflect what is being done (not always 'coding'
- Made office hours options more clear and removed decision making context (as spell team can no longer decide this, it is input instructions from the exec doc)
- Cleaned up initial structural checks at top of checklist (spell description etc)
- Renamed 'Exec Hash' section to 'Spell Description' for better accuracy
- Renamed 'Exec Hash' to 'Exec Doc Hash' for clarity
- Formalized commit anchor generation method
- Enforced a spell naming structure of `TARGET_DATE MakerDAO Executive Spell | Hash: EXEC_DOC_HASH)`
- Fixed `make exec-hash` instructions, which didn't work
- Reworded some items to be binary / boolean checklist items (i.e. checking the box means "Yes, this is true")
- Moved '30 Days expiry'
- More reformatting of Exec Doc section
We can potentially remove `wget` and use `curl` instead for the `description` accompanying comment if we want to, but I won't do that as that changes the spell code (comment) so this needs to be OK'ed by spell participants before it changes. Just noting it here.
- Added sections to add raw URL and Exec Doc Hash when generated to make copy-pasting them less cumbersome and to record what was used when and by whom
- Reformatted how we check exec doc hash (formalize commit anchor fetching method, re-list the steps a reviewer does when checking this and making them checklist items, etc.)
- Added checklist heading item `Developer Environment Checks` (which will be refined shortly)

The-Arbiter commented:
More changes (Working on Goerli checklist for now, will diff and replicate later):

- Updated submodule checks to add a few and mark some as 'non-critical' (i.e. might be broken right now, are not presently a security concern, but are necessary in the checklist for security reasons or otherwise)
- Formalized no 'ZERO' in PCT_RATE naming if Y,Z==0 (i.e. `SIX_PCT_RATE` for 6.00%) (for now...) -> Open to objections / feedback here
- Wording changes in constants section
- Simplified library reinstall method via `rm -r lib && git submodule update --init --recursive` (deletes them for you) -> I didn't do 'foundryup' here too as I don't want to be too invasive for people working with spells. Eventually this can be a script if we want it to be one. The only other footgun protection I want here is a check that `src/DssSpell.sol` is present or something, since I don't want this deleting people's stuff on their machines if they were in the wrong folder. Anyway, that is future work...
- Marked `config.sol` Math Constant checks as non-critical (they'll be caught in test coverage checks OR a PR diff check normally)
- Marked `ds-math` Precision Constant checks as non-critical (it was a check for redundancy with barely any usage in the real world tbh) -> It can be removed once the spell team comes to a consensus on it if we think there are no significant security risks associated with its removal.
- Added formal address declaration rules (for constants AND immutables) which shift review workload earlier on in the checklist (from payments for example, it moves to the constant checking section) -> Good for reviewer mental workflow IMO, open to objections here
- Added formal string declaration rules (for IPFS hashes and the like) based on archive patterns
- Renamed 'dependency checks' to 'library checks'
- Elaborated on chainlog update increment item wording
- Added no unused variable rule (because I removed a similar rule about 'no unused addresses') which will be reworked in a future commit (I think...)

_Please ignore any 'TODO' or similar in the checklist atm as it is a WIP. These will be cleaned up at the end._

Proposals:
- Move Rate constants below Math but above addresses?
- Echo time, some hash, env, etc and env before test runs. This will add to the copy-paste size but not to test run time and will provide transparent info about test execution environments in spell review comments with no overhead for reviewers.

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
````diff
@@ -15,26 +15,23 @@ Spell Actions (Per Exec Sheet):
+  * [ ] Reinstall libraries
+    * [ ] Remove libraries by deleting the `lib` folder
+    * [ ] Install libraries using `git submodule update --init --recursive`
+    ```
+    Insert checked out submodule paths here
+    ```
````
Fix indentation issues:
```suggestion
      ```
      Insert checked out submodule paths here
      ```
```

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
````diff
@@ -33,26 +33,33 @@ Spell Actions (Per Exec Doc):
+  * [ ] Reinstall libraries
+    * [ ] Remove libraries by deleting the `lib` folder
+    * [ ] Install libraries using `git submodule update --init --recursive`
+    ```
+    Insert checked out submodule paths here
+    ```
````
Fix indentation issues:
```suggestion
      ```
      Insert checked out submodule paths here
      ```
```

The-Arbiter commented:
Surprisingly we didn't have any stuff for `line` or `Line`, so I added some scaffolding there.

Indenting for the comment blocks was fixed by committing the suggestion.

The-Arbiter commented:
Issue raised by AmusingAxl:

```
The following mainnet checklist items fail because Governance Facilitators made unrelated commits to the repo:

  * [ ] Search the ['Community' GitHub repo](https://github.com/makerdao/community/tree/master/governance/votes) for the corresponding Exec Doc
  * [ ] Click ['History'](https://github.com/makerdao/community/commits/master/governance/votes/) for the corresponding Exec Doc
```

Commenting here in order to preserve this issue as open for future work on this PR.

Thanks to AmusingAxl for raising this

SidestreamSweatyPumpkin commented on spell/spell-crafter-mainnet-workflow.md:
```diff
@@ -146,7 +155,24 @@ Repo: https://github.com/makerdao/spells-mainnet
 * [ ] Wait for at Least two Approvals to Share for Publishing to GovAlpha
 * [ ] Share Deployed Address in `new-spells`
-* [ ] Fill Spell Crafter Related Boxes in the Main Exec Doc Sheet
+* [ ] Fill Spell Crafter Related Boxes in the Main Exec Sheet
```
Could we move this to the earlier point in the process? New governance team has requested this to be moved to an earlier point in the checklist so that they have confirmations earlier. 

I would suggest moving to the beginning of the list goerli crafter checklist before `Add Spell Actions as per [Exec Sheet]`

SidestreamSweatyPumpkin reviewed (commented):
Desired adjustment based on the recent govops call.

SidestreamColdMelon commented on spell/spell-crafter-mainnet-workflow.md:
```diff
@@ -9,6 +9,11 @@ Repo: https://github.com/makerdao/spells-mainnet
 ![](https://ipfs.io/ipfs/QmUqCvy7c8Qmzn7yZ6D3353wTqCZ3VDAwQKYB37pJ2BjXb)
 
 ### Steps:
+* [ ] (Highly Desired, non-blocking) Make sure that the crafter relatied confirmations in the exec sheet are `Confirmed`
```
I don't think it makes sense to add it to the beginning of the checklist, because first the content needs to be checked that it makes sense from the technical point of view. This is normally done by the end of the development stage, when the code is written, compiled, tests are written and executed. Before that I wouldn't give technical confirm.

Therefore, I propose to remove this disputable point from the PR (but you can submit it with the next PR)

SidestreamColdMelon commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -1,4 +1,4 @@
-# Goerli Executive Spell Review Checklistgoerli
+# Goerli Executive Spell Review Checklist goerli
```
Please remove goerli from the end, because the sentence starts with it

SidestreamColdMelon commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -135,50 +184,77 @@ Spell Actions:
+        * [ ] Accompanying comment above `bump` line in format `// XXM * 1.XX^X * X.XX as a WAD` corresponding to the `val` calculation formula (e.g. `// 15M * 1.03^2 * 1.00 as a WAD`) is present along with the calculation formula on the line above
+        * [ ] IF combining `val` of multiple RWA ilks being combined, `val` calculation is done once per ilk and added to make the total, with workings provided in code comments. The existing `val` value can be retrieved by calling `read()` on `PIP_RWAXX` and converting the result into decimal.
     * [ ] Poke `spotter` to pull in the new price
+  * [ ] Payments
+    * [ ] Payments are not crafted or reviewed on Goerli (comments may be present)
   * [ ] Debt Ceiling (`line`) + Liquidation Oracle Price Bump (`val`)
```
Both `Payments` and `Debt Ceiling` sections should be top-level (but currently are part of the `RWA Updates` section)

SidestreamColdMelon commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -178,16 +178,16 @@ Spell Actions (Per Exec Sheet):
 * [ ] Payments
-* [ ] Payments are not crafted or reviewed on Goerli (comments may be present)
+  * [ ] Payments are not crafted or reviewed on Goerli (comments may be present)
 * [ ] Debt Ceiling (`line`) + Liquidation Oracle Price Bump (`val`)
```
On the second thought this seems to be a subsection related to the `RWA`, so needs to be moved up and be part of the other top-level section, right?

SidestreamSweatyPumpkin replied:
Seems like so. To me makes most sense as well.

Le-Bateleur-2023 commented:
The updates to dev checklist (moving Technical Point confirmations for Exec Vote items up earlier in the process) were requested and agreed to at 12/09/2023 GovOps meeting on Discord. Governance approves these changes.

SidestreamColdMelon commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -1,86 +1,112 @@
-    * [ ] `WAD = 10**18`
-    * [ ] `RAY = 10**27`
-    * [ ] `RAD = 10**45`
+  * [ ] ON
+  * [ ] OFF
```
Nit: I find this change pointless as it doesn't help reviewer check anything, instead I suggest to revert it to the previous text: `On (Collateral Onboarding, Keepers, Integrations, ...)`

SidestreamColdMelon commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -1,86 +1,112 @@
+  * [ ] Timestamp constants used are correct
+    * [ ] Ensure the timestamp [converts](https://www.epochconverter.com/) to the correct date
+    * [ ] Variable naming matches `MONTH_DD_YYYY` (e.g. `MAY_01_2023` for 2023-05-01)
+    * [ ] Time of day makes logical sense (i.e. `11:59:59` for the final day of something, `00:00:00` for the first day of something) in the context of timestamp usage
```
Nit: either specify `PM` or change it to `23:59:59`

SidestreamSweatyPumpkin commented on spell/spell-crafter-mainnet-workflow.md:
```diff
@@ -9,12 +9,13 @@ Repo: https://github.com/makerdao/spells-mainnet
-* [ ] Create Branch on GitHub `PE-<kanban ticket issue number>`
-  * [ ] Ensure the same ticket issue number is used as per `spells-goerli` spell
+* [ ] Create a new branch on the `spells-mainnet` repo named `YYYY-MM-DD` using the initial target date of the spell
+  * [ ] Ensure the same target date is used as the corresponding `spells-goerli` spell branch
```
at this spot i think `goerli` should be used: spells-goerli already has a spell with a date in it, for mainnet the date should also be the same.

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] ensure license is specified
-        * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/conduits/RwaSwapOutputConduit.sol) and it's consistent with previous RWA onboarding
```
We should remove this line. Every deal is different, there's no reason to pinpoint a specific conduit implementation.
```suggestion
```

SidestreamColdMelon replied:
I would just remove the link to the specific hash, but keep the link to the repo with RWAs and wording around "consistent with previous RWA onboarding"

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
+          * [ ] `MCD_ESM` is being relied in the spell (as approved by Governance Facilitators)
+          * [ ] deployer is denied
+        * [ ] no other address has been relied
     * [ ] `RwaUrn2`
```
```suggestion
    * [ ] `RwaUrn2`/`RwaUrn2`
```

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] `outputConduit` matches `RwaOutputConduit<2,3>`
+        * [ ] ensure optimizations are off
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/urns/RwaUrn2.sol) and it's consistent with previous RWA onboardings
```
```suggestion
        * [ ] ensure source matches GitHub code for [`RwaUrn`](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/urns/RwaUrn.sol)/[`RwaUrn2`](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/urns/RwaUrn2.sol) and it's consistent with previous RWA onboardings
```

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
+      * [ ] constructor arg matches [`ChainLog`](https://chainlog.makerdao.com/)
         * [ ] `chainlog`
       * [ ] no `wards`
     * [ ] `RwaInputConduit<2,3>`
```
```suggestion
    * [ ] `Rwa{Swap,}InputConduit{2,3}`
```

SidestreamSweatyPumpkin replied:
`Rwa{Swap,}InputConduit{2,}` is quite cryptic i would say: I'm not sure what is meant here exactly, i have to guess.

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] ensure license is specified
-        * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/conduits/RwaSwapInputConduit2.sol) and it's consistent with previous RWA onboarding
```
```suggestion
        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/tree/8d30ed2cb657641253d45b57c894613e26b4ae1b/src/conduits) and it's consistent with previous RWA onboarding
```

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
-      * [ ] constructor arg matches [chainlog](https://chainlog.makerdao.com/)
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/jars/RwaJar.sol) and it's consistent with previous RWA onboarding
```
```suggestion
        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/8d30ed2cb657641253d45b57c894613e26b4ae1b/src/jars/RwaJar.sol) and it's consistent with previous RWA onboarding
```

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
-        * [ ] ensure license is specified
-        * [ ] ensure source matches github code and it's consistent with previous RWA onboarding
+        * [ ] ensure license is specified (SPDX in code or otherwise)
+        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/conduits/RwaSwapOutputConduit.sol) and it's consistent with previous RWA onboarding
```
```suggestion
        * [ ] ensure source matches [GitHub code](https://github.com/makerdao/rwa-toolkit/tree/8d30ed2cb657641253d45b57c894613e26b4ae1b/src/conduits) and it's consistent with previous RWA onboarding
```

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -94,18 +104,20 @@
         * [ ] `kiss` (whitelist for `urn` destination address `who`)
-     * [ ] Filable
-        * [ ] `quitTo`
+     * [ ] Filable (`RwaOutputConduit` and `RwaInputConduit`)
```
```suggestion
     * [ ] Filable (`RwaSwapOutputConduit` and `RwaSwapInputConduit{,2}`)
```

SidestreamSweatyPumpkin replied:
1. as in https://github.com/makerdao/pe-checklists/pull/12#discussion_r1371306471
2. will adjust to contain `Swap`

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
+        * [ ] `MCD_ESM` is being relied in the spell (as approved by Governance Facilitators)
         * [ ] deployer is denied
         * [ ] no other address has been relied
     * [ ] `RwaOutputConduit<2,3>`
```
```suggestion
    * [ ] `RwaOutputConduit{,2}`/`RwaSwapOutputConduit`
```

SidestreamSweatyPumpkin replied:
as in https://github.com/makerdao/pe-checklists/pull/12#discussion_r1371330713

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
````diff
@@ -1,81 +1,107 @@
+  * [ ] Ensure `description` exactly matches '`Goerli Spell`'
+* [ ] Local Environment Actions
+  * [ ] Update Foundry by running `foundryup`
+  * [ ] Reinstall libraries by running `rm -r lib && git submodule update --init --recursive`
+      ```
+      Insert checked out submodule paths here
+      ```
````
```suggestion
  * [ ] Reinstall libraries by running `rm -r lib && git submodule update --init --recursive`
    ```
    Insert checked out submodule paths here
    ```
```

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -116,69 +152,94 @@ Spell Actions:
+* [ ] Spell actions match the corresponding [Exec Sheet](https://docs.google.com/spreadsheets/d/1w_z5WpqxzwreCcaveB2Ye1PP5B8QAHDglzyxKHG3CHw)
+* [ ] Tests
+  * [ ] Ensure each spell action has sufficient test coverage
+  _List actions for which coverage was checked here_
```
```suggestion
    _List actions for which coverage was checked here_
```

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -116,69 +152,94 @@ Spell Actions:
-* [ ] Local Tests and CI PASS
+  * [ ] Ensure that the `DssExecLib.address` file is not being modified by the spell PR
+  * [ ] Check all CI tests are passing as at the latest commit
+    _Insert most recent commit hash where CI was passing_
```
```suggestion
      _Insert most recent commit hash where CI was passing_
```

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -116,69 +152,94 @@ Spell Actions:
+  * [ ] Check all tests are passing locally using `make test`
+    * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
+
+_Insert your passing local tests here_
```
```suggestion
```
_Insert your passing local tests here_
```
```

amusingaxl replied:
````suggestion
```
_Insert your passing local tests here_
```
````

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -190,21 +251,34 @@ Spell Actions:
+  * [ ] Check all tests are passing locally using `make test`
+    * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
+
+_Insert your passing local tests here_
```
```suggestion
```
_Insert your passing local tests here_
```
```

amusingaxl replied:
````suggestion
```
_Insert your passing local tests here_
```
````

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
````diff
@@ -1,86 +1,111 @@
+  * [ ] Reinstall libraries
+    * [ ] Remove libraries by deleting the `lib` folder
+    * [ ] Install libraries using `git submodule update --init --recursive`
+      ```
+      Insert checked out submodule paths here
+      ```
````
```suggestion
    ```
    Insert checked out submodule paths here
    ```
```

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -91,16 +116,39 @@ Spell Actions:
+          * [ ] [`decreaseGlobalDebtCeiling`](https://github.com/makerdao/dss-exec-lib/blob/c0d3c6c6244468ddab9767de6f853122723fafda/src/DssExecLib.sol#L445C14-L445C39)
+              * [ ] `amount`
+ Autoline Changes
+    * [setIlkAutoLineDebtCeiling](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
```
```suggestion
    * [ ][setIlkAutoLineDebtCeiling](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
```

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -91,16 +116,39 @@ Spell Actions:
+    * [setIlkAutoLineDebtCeiling](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
+      * [ ] `ilk`
+      * [ ] `line`
+    * [setIlkAutoLineParameters](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648)
```
```suggestion
    * [ ][setIlkAutoLineParameters](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648)
```

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -86,16 +112,26 @@ Spell Actions:
+      * [ ] [`setGlobalDebtCeiling`](https://github.com/makerdao/dss-exec-lib/blob/c0d3c6c6244468ddab9767de6f853122723fafda/src/DssExecLib.sol#L428) or [`increaseGlobalDebtCeiling`](https://github.com/makerdao/dss-exec-lib/blob/c0d3c6c6244468ddab9767de6f853122723fafda/src/DssExecLib.sol#L436) or [`decreaseGlobalDebtCeiling`](https://github.com/makerdao/dss-exec-lib/blob/c0d3c6c6244468ddab9767de6f ...
+          * [ ] `amount`
+  * [ ] Autoline Changes
+    * [setIlkAutoLineDebtCeiling](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
```
```suggestion
    * [ ][setIlkAutoLineDebtCeiling](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
```

amusingaxl commented on spell/spell-reviewer-goerli-checklist.md:
```diff
@@ -86,16 +112,26 @@ Spell Actions:
+    * [setIlkAutoLineDebtCeiling](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
+      * [ ] `ilk`
+      * [ ] `line`
+    * [setIlkAutoLineParameters](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648)
```
```suggestion
    * [ ] [setIlkAutoLineParameters](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648)
```

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -121,69 +169,141 @@ Spell Actions:
+  * [ ] Check all tests are passing locally using `make test`
+    * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
+
+_Insert your passing local tests here_
```
````suggestion
```
_Insert your passing local tests here_
```
````

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -121,69 +169,141 @@ Spell Actions:
+* [ ] Spell actions match the corresponding Exec Doc
+* [ ] Tests
+  * [ ] Ensure each spell action has sufficient test coverage
+  _List actions for which coverage was checked here_
```
```suggestion
    _List actions for which coverage was checked here_
```

amusingaxl commented on spell/spell-reviewer-mainnet-checklist.md:
```diff
@@ -195,16 +315,31 @@ Spell Actions:
+  * [ ] Check all tests are passing locally using `make test`
+    * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
+
+_Insert your passing local tests here_
```
````suggestion
```
_Insert your passing local tests here_
```
````

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
         * [ ] deployer is denied
         * [ ] no other address has been relied
-    * [ ] `RwaOutputConduit<2,3>`
+    * [ ] `RwaOutputConduit<2,3>`/`RwaSwapOutputConduit`
```
We don't have `RwaOutputConduit3`, it was renamed to `RwaSwapOutputConduit`.
```suggestion
    * [ ] `RwaOutputConduit<2>`/`RwaSwapOutputConduit`
```

amusingaxl commented on spell/rwa-onboarding-checklist.md:
```diff
@@ -11,64 +12,73 @@
+      * [ ] constructor arg matches [`ChainLog`](https://chainlog.makerdao.com/)
         * [ ] `chainlog`
       * [ ] no `wards`
     * [ ] `RwaInputConduit<2,3>`
```
```suggestion
    * [ ] `RwaInputConduit<2>` / `RwaSwapInputConduit<2>`
```

amusingaxl reviewed (approved):
Let's get this done!
